packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/Carousel/CarouselItem.tsx

```diff
@@ -0,0 +1,25 @@
+import { useContextSelector } from '@fluentui/react-context-selector';
+import { CarouselContext } from './useCarouselCollection';
+import * as React from 'react';
+import { CarouselItemProps } from './Carousel.types';
+
+// TODO: We are going to move this to it's own <CarouselItem/> component, for now we use a simple function to wrap.
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export function CarouselItem_unstable(props: CarouselItemProps) {
+  const { value } = props;
+
+  const carouselContext = useContextSelector(CarouselContext, c => c);
+
+  if (!carouselContext) {
+    return props.children;
+  }
+
+  const visible = carouselContext.value === value;
+
+  return (
+    <>
+      <span data-carousel-item={value} data-carousel-active-item={!!visible} hidden />
```

Comment:
```suggestion
      <span {...{[CAROUSEL_ITEM]: value, [CAROUSEL_ACTIVE_ITEM]: !!visible}} hidden />
```

Reply:
Also instead of rendering span, you can render div with children being passed dependeing on visibility - it will make `forwardRef` handling much easier

Reply:
@ling1726 as I recall, having an always rendered `span` was simplification as then the same element is always present.

Reply:
@layershifter by changing to `div` the same element can be present always right? it would just render children or not

Reply:
that works 👍